libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/marker/FreeDriveLongPressMapComponent.kt

```diff
@@ -0,0 +1,46 @@
+package com.mapbox.navigation.dropin.component.marker
+
+import com.mapbox.maps.MapView
+import com.mapbox.maps.plugin.gestures.OnMapLongClickListener
+import com.mapbox.maps.plugin.gestures.gestures
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.component.destination.DestinationAction
+import com.mapbox.navigation.dropin.component.destination.DestinationViewModel
+import com.mapbox.navigation.dropin.component.navigation.NavigationState
+import com.mapbox.navigation.dropin.component.navigation.NavigationStateAction
+import com.mapbox.navigation.dropin.component.navigation.NavigationStateViewModel
+import com.mapbox.navigation.dropin.component.routefetch.RoutesAction
+import com.mapbox.navigation.dropin.component.routefetch.RoutesViewModel
+import com.mapbox.navigation.dropin.lifecycle.UIComponent
+import com.mapbox.navigation.dropin.model.Destination
+
+internal class FreeDriveLongPressMapComponent(
+    private val mapView: MapView,
+    private val navigationStateViewModel: NavigationStateViewModel,
+    private val routesViewModel: RoutesViewModel,
+    private val destinationViewModel: DestinationViewModel,
+) : UIComponent() {
+
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        super.onAttached(mapboxNavigation)
+        mapView.gestures.addOnMapLongClickListener(longClickListener)
+    }
+
+    override fun onDetached(mapboxNavigation: MapboxNavigation) {
+        super.onDetached(mapboxNavigation)
+        mapView.gestures.removeOnMapLongClickListener(longClickListener)
+    }
+
+    private val longClickListener = OnMapLongClickListener { point ->
+        destinationViewModel.invoke(
+            DestinationAction.SetDestination(Destination(point))
+        )
+        routesViewModel.invoke(
+            RoutesAction.SetRoutes(emptyList())
+        )
+        navigationStateViewModel.invoke(
+            NavigationStateAction.Update(NavigationState.DestinationPreview)
+        )
```

Comment:
Before we had a [single component](https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/navigation/NavigationStateComponent.kt) that centralized NavigationState update logic in a single class, and was driven by 3 sources of truth: `destination`, `routesList` and `navigationStarted`.

In this PR we remove it and scatter the update logic throughout multiple UIComponents.

This worries me.  I think this change will make our code more error prone, difficult to maintain and debug.

Reply:
Honestly spent a bit of time working on the centralized approach. I was having issues getting it to work, because we need different behaviors in different states. It's not often we can, "always do X". Decoupling components give some freedom to fix issues in isolation.

Some examples:
 - When we start the trip session, replay automatically starts. We want to disable audio guidance for all states except active navigation.
 - When we start the trip session, audio guidance starts. We want to disable audio guidance for all states except active navigation.
 - The route is only shown for the route preview. 
 - The back stack logic was crazy. it was conditionally deciding how to behave based on many states. The back stack behavior is different from the long press behavior
 - The stop navigation button needs to behave different from the back stack and the long press

Reply:
The `RoutesViewModel` was also starting/stopping the trip session. But we need a centralized place to decide if replay will work. There were lots of test cases to handle when trying to handle replay https://github.com/mapbox/navigation-sdks/issues/1549

The `RoutesViewModel`, already has complex problems to solve. If we add trip session management to an issue that it needs to be involved in, we have more issues.

Here are a couple issues for RoutesViewModel
 - Handle waypoints
 - Handle canceling current Fetching route when new route is Fetched

Reply:
> Before we had a [single component](https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/navigation/NavigationStateComponent.kt) that centralized NavigationState update logic in a single class, and was driven by 3 sources of truth: destination, routesList and navigationStarted.

Realizing I'm mentioning a lot of other issues, but not directly addressing this one. 

The `NavigationComponent` was introduced to resolve a circular dependency with ViewModels. But I can see the desire to delete the `NaviationStateAction.Update` from individual components.

This is a potential solution for guaranteeing the ViewModel state and NavigationState are in sync and reactive. I'll bring that part back and see how it behaves

Reply:
Remembered again what the issues were when bringing it back. Cutting them as tickets as follow ups